pkg/proxy/iptables/proxier.go

```diff
@@ -1628,17 +1628,25 @@ func openLocalPort(lp *utilproxy.LocalPort) (utilproxy.Closeable, error) {
 	var socket utilproxy.Closeable
 	switch lp.Protocol {
 	case "tcp":
-		listener, err := net.Listen("tcp", net.JoinHostPort(lp.IP, strconv.Itoa(lp.Port)))
+		network := "tcp4"
+		if isIPv6 {
+			network = "tcp6"
+		}
+		listener, err := net.Listen(network, net.JoinHostPort(lp.IP, strconv.Itoa(lp.Port)))
```

Comment:
it'd be a little simpler to just have like `family = "4"` / `family = "6"` outside of the switch and then `net.Listen("tcp" + family, ...`. (So there's less duplication between the tcp and udp cases.)